fix(api): Implement Flex Stacker deck conflict checking

CaseyBatten opened the pull request:
# Overview
Covers RQA-4539

The deck conflict checker is a circus and we play the role of it's clowns.

We missed implementing the deck conflict checker for the Flex Stacker module, which previously just returned None which proceeded to boot it out of further deck conflict checking. This lead to issues like trash bins being loadable into A3 when a stacker is in A4.

## Test Plan and Hands on Testing
- [x] Implement Unit Test to enforce stacker deck conflict check

- [x] The following protocol should fail analysis with this error: "DeckConflictError [line 20]: flexStackerModuleV1 in slot B3 prevents trash bin from using slot B3."
```
from opentrons.protocol_api import ProtocolContext

metadata = {
    "protocolName": "Flex Stacker Trashbin Error Protocol",
    "author": "Opentrons <protocols@opentrons.com>",
}
requirements = {
    "robotType": "Flex",
    "apiLevel": "2.25",
}
def run(protocol: ProtocolContext) -> None:
    """Protocol."""
    stacker_B = protocol.load_module( "flexStackerModuleV1", "B4")
    stacker_B.set_stored_labware(
        load_name="opentrons_flex_96_tiprack_200ul",
        count=1,
        lid="opentrons_flex_tiprack_lid",
    )

    trash = protocol.load_trash_bin("B3")

    tiprack = stacker_B.retrieve()
    protocol.move_lid(tiprack, trash, use_gripper=True)
```

## Changelog


## Review requests
So we wrap the Flex Stacker conflict errors in the "other module" class, which is correct for now and kind of makes checking for it at all as opposed to just returning under the "else" conditional redundant. Do we want to leave it that way incase in the future we want a more fleshed out stacker conflict error? Or should we just excise it entirely?

## Risk assessment
Low - fixes existing behavior

CaseyBatten committed:
implement stacker deck conflict checking

CaseyBatten committed:
include stacker in deck conflict tests

sfoster1 reviewed (approved):
Lol

sfoster1 reviewed (changes requested):
just noticed the snapshot failures, definitely something wrong

sfoster1 committed:
fix: properly handle stacker in wrapped tests

See, this is the part where for me it started to look like a problem. You know, I wanted to sacrifice the Precentral vein to order to get some exposure, but because of this guy's normal variation I got excited and all of a sudden I didn't know whether if I was looking at the Precentral vein or one of the internal cerebral veins or the vein of Galen or the Basilar vein of Rosenthal. So on my own, me, at this point I was ready to say, "That's it. Let's get out."

sfoster1 committed:
fix: lady macbeth vibes

So if you let other stuff "share" the "same" "slot" that a stacker is
in, it turns out that later on the engine core machinery is going to try
to put two things in the same dict slot, which it doesn't want to do. To
make it do that we're going to make a combo object, which is pretty
cool, and definitely doesn't make me suspect Banquo's ghost is going to
show up in the PR comments

sfoster1 committed:
fix: add some "lovely" tests

sfoster1 reviewed (approved):
well, it's as good as it's gonna get

sfoster1 committed:
random probably wasn't smart

SyntaxColoring reviewed (approved):
Brutal, but I think I understand why the brutality is necessary.
